pyqs/worker.py

```diff
@@ -267,6 +267,164 @@ def _run_hooks(self, hook_name, context):
             hook(context)
 
 
+class SimpleProcessWorker(BaseWorker):
+
+    def __init__(self, queue_url, batchsize,
+                 connection_args=None, *args, **kwargs):
+        super(SimpleProcessWorker, self).__init__(*args, **kwargs)
+        if connection_args is None:
+            connection_args = {}
+        self.connection_args = connection_args
+        self.queue_url = queue_url
+
+        sqs_queue = get_conn(**self.connection_args).get_queue_attributes(
+            QueueUrl=queue_url, AttributeNames=['All'])['Attributes']
+        self.visibility_timeout = int(sqs_queue['VisibilityTimeout'])
+
+        self.batchsize = batchsize
+        self._messages_to_process_before_shutdown = 100
+        self.messages_processed = 0
+
+    def run(self):
+        # Set the child process to not receive any keyboard interrupts
+        signal.signal(signal.SIGINT, signal.SIG_IGN)
+
+        logger.info(
+            "Running SimpleProcessWorker: {}, pid: {}".format(
+                self.queue_url, os.getpid()))
+
+        while not self.should_exit.is_set() and self.parent_is_alive():
+            self.read_message()
+
+            if self.messages_processed \
+                    >= self._messages_to_process_before_shutdown:
+                self.shutdown()
+
+    def read_message(self):
+        messages = self._get_connection().receive_message(
+            QueueUrl=self.queue_url,
+            MaxNumberOfMessages=self.batchsize,
+            WaitTimeSeconds=LONG_POLLING_INTERVAL,
```

Comment:
Do we want to keep the `LONG_POLLING_INTERVAL`. That means it could take 20 seconds after writing an item to SQS to pick it up. It would be nice if this was configurable or if the wait time was 0 if batchsize = 1. It's more API calls but I don't think that matters much

Reply:
This is already an issue #60 and yes maybe we can make it configurable, but this would generally apply to `SimpleProcessWorker` or `ReadWorker`, so maybe we can deal with it in a separate PR.